winit/src/platform_impl/linux/mod.rs

```diff
@@ -169,6 +170,13 @@ impl EventLoop {
     pub fn window_target(&self) -> &dyn ActiveEventLoop {
         x11_or_wayland!(match self; EventLoop(evlp) => evlp.window_target())
     }
+
+    pub fn main_drm_device(&self) -> Option<dev_t> {
+        match self {
+            EventLoop::Wayland(evlp) => evlp.main_drm_device(),
+            EventLoop::X(_) => None,
```

Comment:
I don't care enough about X to implement this. But it should be possible by using the `Dri3Open` request to get a DRM fd and then calling stat on it.